qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py

```diff
@@ -0,0 +1,701 @@
+        with open(result_filepath, 'rt', encoding='utf-8') as file:
+            result_string = file.read().replace('\n', '')
+            for loop_histogram in histogram_data_files:
+                where_loop_histogram = result_string.rfind(loop_histogram)
+                if where_loop_histogram != -1:
+                    where_loop_histogram_wR = result_string.find('Final refinement wR =', where_loop_histogram)
+                    if where_loop_histogram_wR != -1:
+                        where_loop_histogram_wR_end = result_string.find('%', where_loop_histogram_wR)
+                        logger.notice(loop_histogram)
+                        logger.notice(result_string[where_loop_histogram_wR: where_loop_histogram_wR_end + 1])
+
+    def get_crystal_params_from_instrument(self, instrument):
+        crystal_params = []
+        if not self.number_of_regions:
+            crystal_params = self.find_in_file(instrument, "ICONS", "S", "INS", strip_separator="ICONS\t")
+
+        else:
+            for region_index in range(1, self.number_of_regions+1):
+                loop_crystal_params = self.find_in_file(instrument, f"{region_index} ICONS", "S", "INS",
+                                                        strip_separator="ICONS\t")
+                crystal_params.append(loop_crystal_params)
+        tof_min = []
+        for loop_crystal_param_string in crystal_params:
+            list_crystal_params = loop_crystal_param_string.split("\t")
+            if len(list_crystal_params):
+                dif_c = float(list_crystal_params[0] or 0.0)
+                dif_a = float(list_crystal_params[1] or 0.0)
+                t_zero = float(list_crystal_params[2] or 0.0)
+                tof_min.append((dif_c * self.dSpacing_min) + (dif_a * (self.dSpacing_min ** 2)) + t_zero)
+        return tof_min
+
+    # ===================
+    # Pawley Reflections
+    # ===================
+
+    def choose_cell_lengths(self, phase_file_path: str):
+        if self.override_cell_length_string:
+            if "," not in self.override_cell_length_string:
+                overriding_cell_lengths_list = [float(self.override_cell_length_string)] * 3
+            else:
+                overriding_cell_lengths_list = [float(x) for x in self.override_cell_length_string.split(",")]
+
+            if len(overriding_cell_lengths_list) != 3:
+                logger.error(f"The number of Override Cell Length values ({len(overriding_cell_lengths_list)}) "
+                             f"must be 1 or 3 (and separated by commas).")
+                return None
+
+            cell_lengths = " ".join([str(overriding_cell_lengths_list[0]),
+                                     str(overriding_cell_lengths_list[1]),
+                                     str(overriding_cell_lengths_list[2])])
+        else:
+            cell_length_a = self.find_in_file(phase_file_path, '_cell_length_a', ' ', '_cell_length_b')
```

Comment:
If an invalid phase file is supplied these calls to `find_in_file` will return None so please can you check for None and return an "invalid phase file format" error or similar? The join command on L411 falls over if any of the cell_length_x variables are None